Fix copy/paste error and duplicate config (#4406)
viamrobotics/docs: docs/manage/reference/viam-agent/_index.md

      "offline_before_starting_hotspot_minutes": 2,
      "user_idle_minutes": 5,
      "retry_connection_timeout_minutes": 10,

>       "turn_on_hotspot_if_wifi_has_no_internet": true,

Author: I thought it would be better to use `true` as the value since this example specifies `additional_networks`, and that config doesn't matter unless this is set to true. Personally I don't see the value of an example of default values but maybe I'm missing something?

| `hotspot_prefix` | string | Optional | `viam-agent` will prepend this to the hostname of the device and use the resulting string for the provisioning hotspot SSID. Default: `"viam-setup"`. |
| `manufacturer` | string | Optional | Purely informative. May be displayed on captive portal or provisioning app. Default: `"viam"`. |
| `model` | string | Optional | Purely informative. May be displayed on captive portal or provisioning app. Default: `"custom"`. |

> | `offline_before_starting_hotspot_minutes` | integer | Optional | Amount of time the device will spend searching for the specified Wi-Fi network, in minutes, before the machine begins searching for an alternate network or broadcasting a wireless hotspot. Default: `2`. |

Author: ```suggestion
| `offline_before_starting_hotspot_minutes` | integer | Optional | Amount of time the device will spend connected to a Wi-Fi network that is not connected to the Internet, in minutes, before the machine connects to an alternate network or begins broadcasting a wireless hotspot. Default: `2`. |
```
Is this correct?

> | `offline_before_starting_hotspot_minutes` | integer | Optional | Amount of time the device will spend searching for the specified Wi-Fi network, in minutes, before the machine begins searching for an alternate network or broadcasting a wireless hotspot. Default: `2`. |

Reviewer: Not quite... the timeout doesn't affect when it starts searching for other networks. Let me know what you think of this wording:
```suggestion
| `offline_before_starting_hotspot_minutes` | integer | Optional | Amount of time the device will spend searching/waiting for a connection (after detecting itself as offline) before entering provisioning (hotspot) mode. Default: `2`. |
```

> | `offline_before_starting_hotspot_minutes` | integer | Optional | Amount of time the device will spend searching for the specified Wi-Fi network, in minutes, before the machine begins searching for an alternate network or broadcasting a wireless hotspot. Default: `2`. |

Author: I updated my suggestion above to use the 'offline' term, just like in the `turn_on_hotspot_if_wifi_has_no_internet` description, since it's a helpful vocabulary term for this situation. I'd also like to avoid parentheticals and the verbiage 'entering provisioning mode' since they make it harder for folks to understand what's going on here if they aren't already really familiar with the platform. Does the updated snippet look correct to you?

> | `offline_before_starting_hotspot_minutes` | integer | Optional | Amount of time the device will spend searching for the specified Wi-Fi network, in minutes, before the machine begins searching for an alternate network or broadcasting a wireless hotspot. Default: `2`. |

Reviewer: "before the machine connects to an alternate network or begins broadcasting a wireless hotspot" the first part of that is what I was reading. It implies that the timeout has to pass before other networks will be tried, which is what I'm saying is incorrect. The time value set here only affects how long connectivity can be "offline" before starting the hotspot. The entire time it is offline, it should be trying other networks (if possible.)

> | `offline_before_starting_hotspot_minutes` | integer | Optional | Amount of time the device will spend searching for the specified Wi-Fi network, in minutes, before the machine begins searching for an alternate network or broadcasting a wireless hotspot. Default: `2`. |

Reviewer: I see now you edited your comment above. The edited version now feels correct. That is, the one worded: "Amount of time the device will spend offline, in minutes, before the machine begins broadcasting a wireless hotspot."

Lastly, FYI, the reason I said "enter provisioning mode" is that it's not JUST hotspot/wifi involved. We now support bluetooth provisioning, and the wifi/hotspot side of things can be disabled entirely even, but the rules here still apply to the provisioning mode in general (not just the "hotspot".) That said, keeping it worded as above (with the "hotspot" language) is probably fine for now.

| `retry_connection_timeout_minutes` | integer | Optional | Provisioning mode will exit after this time (in minutes), to allow other unmanaged (for example wired) or manually configured connections to be tried. Provisioning mode will restart if the connection/online status doesn't change. Default: `10`. |

> | `turn_on_hotspot_if_wifi_has_no_internet` | boolean | Optional | When enabled, Wi-Fi connections without Internet access are considered offline. After `offline_before_starting_hotspot_minutes` minutes offline, your device will begin broadcasting a hotspot. Default: `false`. |

Author: ```suggestion
| `turn_on_hotspot_if_wifi_has_no_internet` | boolean | Optional | When enabled, Wi-Fi connections without Internet access are considered offline. After `offline_before_starting_hotspot_minutes` minutes offline, your device will begin broadcasting a hotspot. This setting must be enabled for your device to attempt connecting to `additional_networks`. Default: `false`. |
```

> | `turn_on_hotspot_if_wifi_has_no_internet` | boolean | Optional | Controls whether or not a Wi-Fi connection without Internet access is considered **online**. When enabled, Wi-Fi connections without Internet access are considered **offline**. After `offline_before_starting_hotspot_minutes` minutes offline, your device will begin broadcasting a hotspot. Default: `false` (Wi-Fi connections without Internet access are considered online). |

Reviewer 2: ```suggestion
| `turn_on_hotspot_if_wifi_has_no_internet` | boolean | Optional | When enabled, Wi-Fi connections without Internet access are considered offline. After `offline_before_starting_hotspot_minutes` minutes offline, your device will begin broadcasting a hotspot. Default: `false` (Wi-Fi connections without Internet access are considered online). |
```
I think you could remove this since it's basically repeated, and because the attribute name helps cover it too

Also [opt] you *could* remove the parenthetical for the default behavior

> | `turn_on_hotspot_if_wifi_has_no_internet` | boolean | Optional | Controls whether or not a Wi-Fi connection without Internet access is considered 'online'. If enabled, if the Wi-Fi network provided during initial provisioning does not provide a connection to the Internet, the device will wait `offline_before_starting_hotspot_minutes` minutes. If the Wi-Fi network still does not provide a connection to the internet after this time has passed, your device will search for other Wi-Fi networks configured in `additional_networks`. If those networks cannot be found and the configuration includes a hotspot, the device will begin hosting a hotspot. Once a device connects to an alternate network or begins hosting a hotspot, the device will not attempt to reconnect to the original Wi-Fi network provided during intial provisioning until after a reboot. Default: `false`. |

Author: ```suggestion
| `turn_on_hotspot_if_wifi_has_no_internet` | boolean | Optional | Controls whether or not a Wi-Fi connection without Internet access is considered 'online'. If enabled, when connected to an 'offline' network, the device will wait `offline_before_starting_hotspot_minutes` minutes before connecting to another Wi-Fi network configured in `additional_networks`. If none of those networks provide an 'online' connection, the device will begin hosting a hotspot, if your configuration includes one. Once your device begins hosting a hotspot, you must reboot to connect to another network. Default: `false`. |
```

> | `turn_on_hotspot_if_wifi_has_no_internet` | boolean | Optional | Controls whether or not a Wi-Fi connection without Internet access is considered 'online'. If enabled, if the Wi-Fi network provided during initial provisioning does not provide a connection to the Internet, the device will wait `offline_before_starting_hotspot_minutes` minutes. If the Wi-Fi network still does not provide a connection to the internet after this time has passed, your device will search for other Wi-Fi networks configured in `additional_networks`. If those networks cannot be found and the configuration includes a hotspot, the device will begin hosting a hotspot. Once a device connects to an alternate network or begins hosting a hotspot, the device will not attempt to reconnect to the original Wi-Fi network provided during intial provisioning until after a reboot. Default: `false`. |

Reviewer: Again, that offline_before... timeout doesn't control the timing of when it tries other networks, only when it's DONE with that and enters hotspot mode. If the "online" test fails, it always moves onto trying the next network.

I'm also unclear where you got the concept "the device will begin hosting a hotspot, if your configuration includes one." There's no need to include a hotspot config.

Likewise, I'm confused why you say you must reboot to connect to another network. There's no such requirement at all. At most, that was a troubleshooting step for some devices with a since-fixed bug.)

> | `turn_on_hotspot_if_wifi_has_no_internet` | boolean | Optional | Controls whether or not a Wi-Fi connection without Internet access is considered 'online'. If enabled, if the Wi-Fi network provided during initial provisioning does not provide a connection to the Internet, the device will wait `offline_before_starting_hotspot_minutes` minutes. If the Wi-Fi network still does not provide a connection to the internet after this time has passed, your device will search for other Wi-Fi networks configured in `additional_networks`. If those networks cannot be found and the configuration includes a hotspot, the device will begin hosting a hotspot. Once a device connects to an alternate network or begins hosting a hotspot, the device will not attempt to reconnect to the original Wi-Fi network provided during intial provisioning until after a reboot. Default: `false`. |

Author: ```suggestion
| `turn_on_hotspot_if_wifi_has_no_internet` | boolean | Optional | Controls whether or not a Wi-Fi connection without Internet access is considered **online**. When enabled, Wi-Fi connections without Internet access are considered **offline**. After `offline_before_starting_hotspot_minutes` minutes offline, your device will begin broadcasting a hotspot. Default: `false` (Wi-Fi connections without Internet access are considered online). |
```
Gotcha, removed the bits about 'if your configuration includes one' since there are defaults for that and streamlined the definition to better explain the online/offline distinction.

If I may ask, how does an SBC ever scan for other Wi-Fi networks while hosting a hotspot? I've never worked with an SBC that can both broadcast a hotspot and scan for networks simultaneously, so I think I must be misunderstanding what's going on there.
